stdlib/src/mmap.rs

```diff
@@ -0,0 +1,1053 @@
+            if self.closed() {
+                return Ok(());
+            }
+
+            if self.exports.load() > 0 {
+                return Err(vm.new_buffer_error("cannot close exported pointers exist.".to_owned()));
+            }
+            self.closed.store(true);
+            *self.mmap.lock() = None;
+
+            Ok(())
+        }
+
+        fn get_find_range(&self, options: FindOptions) -> (usize, usize) {
+            let pos = self.inner_pos();
+            let size = self.inner_size();
+            let start = options.start.unwrap_or(pos);
+            let end = options.end.unwrap_or(size);
+
+            (
+                saturate_index(start, size.try_into().unwrap()),
+                saturate_index(end, size.try_into().unwrap()),
+            )
+        }
+
+        #[pymethod]
+        fn find(&self, options: FindOptions, vm: &VirtualMachine) -> PyResult<isize> {
+            let (start, end) = self.get_find_range(options.clone());
+
+            let sub = &options.sub;
+
+            if sub.is_empty() {
+                return Ok(0);
+            }
+            let pos = match self.check_valid(vm)?.deref().as_ref().unwrap() {
+                MmapObj::Read(mmap) => mmap[start..end]
+                    .windows(sub.len())
+                    .position(|window| window == sub),
+                MmapObj::Write(mmap) => mmap[start..end]
+                    .windows(sub.len())
+                    .position(|window| window == sub),
+            };
+
+            if let Some(i) = pos {
+                return Ok((start + i) as isize);
+            }
+
+            Ok(-1)
+        }
+
+        #[pymethod]
+        fn rfind(&self, options: FindOptions, vm: &VirtualMachine) -> PyResult<PyIntRef> {
+            let (start, end) = self.get_find_range(options.clone());
+
+            let sub = &options.sub;
+            if sub.is_empty() {
+                return Ok(PyInt::from(0usize).into_ref(vm));
+            }
+
+            let pos = match self.check_valid(vm)?.deref().as_ref().unwrap() {
+                MmapObj::Read(mmap) => mmap[start..end]
+                    .windows(sub.len())
+                    .rposition(|window| window == sub),
+                MmapObj::Write(mmap) => mmap[start..end]
+                    .windows(sub.len())
+                    .rposition(|window| window == sub),
+            };
+
+            if let Some(i) = pos {
+                return Ok(PyInt::from(start + i).into_ref(vm));
+            }
+
+            Ok(PyInt::from(-1).into_ref(vm))
+        }
+
+        #[pymethod]
+        fn flush(&self, options: FlushOptions, vm: &VirtualMachine) -> PyResult<()> {
+            let offset = options.offset.unwrap_or(0);
+            let size = options.size.unwrap_or_else(|| self.inner_size());
+
+            if size < 0 || offset < 0 || self.inner_size() - offset < size {
+                return Err(vm.new_value_error("flush values out of range".to_owned()));
+            }
```

Comment:
```rust
let offset = offset as usize;
let size = size as usize;
```
Let's move this casting here to enhance locality. Because they are safe due to upper checking.